api_docs/construct-narrow.md

```diff
@@ -84,6 +90,24 @@ filters did.
 
 ### Message IDs
 
+The `with` operator uses message ID, which must be encoded as a string
+for its operand.
```

Comment:
This should accept the ID as a number, not only as a string. That is, it should accept `"operand":12345`, not only `"operand":"12345"`.

The use of a string here is kind of messy, and ideally I'd say it would *only* accept a number, not a string. But IIUC getting the web client to send a number instead of a string would be a nontrivial refactor, and I'm fine with leaving that out of scope for this issue and having the server accept a string as a workaround for that behavior of the web client.

The example below contradicts this description — it has the operand as a number after all.

Reply:
updated.